packages/sg1/src/lib.rs

```diff
@@ -1,37 +1,60 @@
-use cosmwasm_std::{coins, BankMsg, CosmosMsg, Decimal, MessageInfo, Uint128};
+use cosmwasm_std::{coins, Addr, BankMsg, CosmosMsg, Decimal, MessageInfo, Uint128};
 use cw_utils::{must_pay, PaymentError};
 use sg_std::{create_fund_community_pool_msg, StargazeMsgWrapper, NATIVE_DENOM};
 use thiserror::Error;
 
 // governance parameters
 const FEE_BURN_PERCENT: u64 = 50;
+const DEV_INCENTIVE_PERCENT: u64 = 10;
 
 type SubMsg = CosmosMsg<StargazeMsgWrapper>;
 
 /// Burn and distribute fees and return an error if the fee is not enough
-pub fn checked_fair_burn(info: &MessageInfo, fee_amount: u128) -> Result<Vec<SubMsg>, FeeError> {
+pub fn checked_fair_burn(
+    info: &MessageInfo,
+    fee: u128,
+    developer: Option<Addr>,
+) -> Result<Vec<SubMsg>, FeeError> {
     let payment = must_pay(info, NATIVE_DENOM)?;
-    if payment.u128() < fee_amount {
-        return Err(FeeError::InsufficientFee(fee_amount, payment.u128()));
+    if payment.u128() < fee {
+        return Err(FeeError::InsufficientFee(fee, payment.u128()));
     };
 
-    Ok(fair_burn(fee_amount))
+    Ok(fair_burn(fee, developer))
 }
 
 /// Burn and distribute fees, assuming the right fee is passed in
-pub fn fair_burn(fee_amount: u128) -> Vec<SubMsg> {
-    // calculate the fee to burn
+pub fn fair_burn(fee: u128, developer: Option<Addr>) -> Vec<SubMsg> {
+    let mut msgs: Vec<SubMsg> = vec![];
+
+    let (burn_percent, dev_fee) = match developer {
+        Some(dev) => {
+            let dev_fee = (Uint128::from(fee) * Decimal::percent(DEV_INCENTIVE_PERCENT)).u128();
+            let msg = BankMsg::Send {
+                to_address: dev.to_string(),
+                amount: coins(dev_fee, NATIVE_DENOM),
+            };
+            msgs.push(SubMsg::Bank(msg));
+            (
+                Decimal::percent(FEE_BURN_PERCENT - DEV_INCENTIVE_PERCENT),
```

Comment:
what if `DEV_INCENTIVE_PERCENT > FEE_BURN_PERCENT`

Reply:
This should be enforced in the governance layer (when its implemented). Right now its fine since the parameters are hardcoded.